core/api/src/main/java/com/alipay/sofa/rpc/client/ProviderInfo.java

```diff
@@ -461,6 +427,15 @@ public int getWeight() {
         return weight;
     }
 
+    /**
+     * Get the origin weight
+     *
+     * @return the weight
+     */
+    public int getOriginWeight() {
```

Comment:
The weight is mutable, this method is no need here. If somewhere need to get origin weight of provider, just use `getStaticAttr("weight")`